### ISSUE TITLE
Neogit showing incorrect/duplicate data in status buffer

### ISSUE THREAD

**Issue by milkias17** [NeogitOrg/neogit#1384]
### Description

For some reason upon staging a file, neogit upon updating the status buffer shows multiple entries of each file.

### Neovim version

NVIM v0.10.0
Build type: RelWithDebInfo
LuaJIT 2.1.1713773202

### Operating system and version

Linux fedora 6.8.11-300.fc40.x86_64

### Steps to reproduce

1. nvim
2. :Neogit <CR>
3. Stage files (sometimes it comes up with this situation)

### Actual behavior

![image](https://github.com/NeogitOrg/neogit/assets/66522725/1858ac25-5993-47b4-ad7d-7575752728e0)

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({
	kind = "split",
	disable_commit_confirmation = true,
	disable_builtin_notifications = true,
	disable_insert_on_commit = "auto",
	commit_editor = {
		kind = "split",
	},
	graph_style = "unicode",

}) -- For instance, setup Neogit
```

**Comment by CKolkey** [comment]
Thanks - I know about this one, just haven't had time lately to look into it

**Comment by ErnestChiu2023** [comment]
fwiw this doesn't happen on the nightly branch from what I've seen

**Comment by CKolkey** [comment]
No, it wouldn't. It's a recent thing, and nightly hasn't been updated in a while.

**Comment by viocost** [comment]
Also experiencing this one

**Comment by oneart-dev** [comment]
same issue. Also neogit start to be super laggy for some reason. Staging entire files with small changes can take seconds

**Comment by ktsujister** [comment]
This seems to be fixed with #1386.

**Comment by CKolkey** [comment]
I suspected - thanks for confirming it :)

**Comment by CKolkey** [comment]
> same issue. Also neogit start to be super laggy for some reason. Staging entire files with small changes can take seconds

@oneart-dev If you want to make a profile with the lag you're talking about, or provide a file as an example, I'd appreciate it :)

You can use plenary to make the profile

`require("plenary.profile").start("profile.log", { flame = true })`
`require("plenary.profile").stop`

**Comment by GCBallesteros** [comment]
@CKolkey I moved to the nightly branch because of the extreme lag being reported by @oneart-dev . I will move back to master and run that profile if I still encounter that behavior.

**Comment by CKolkey** [comment]
Thanks - I try hard to prevent performance issues, and don't encounter this one. So, any logs/profiling is great

**Comment by oneart-dev** [comment]
@CKolkey fix working flawlessly. No lags, no duplicates. Great job!
